### ISSUE TITLE
Blockified Add to Cart with Options block: Grouped Product Title should be of level 3 when in the Single Product block

### ISSUE THREAD

**Issue by Aljullu** [woothemes/woocommerce#56068]
* Blocked by https://github.com/WordPress/gutenberg/issues/30549 (see the [comment below](https://github.com/woocommerce/woocommerce/issues/56068#issuecomment-2704407810) for more details).

When using the Single Product block to display a Grouped product, we should make it so we have the correct hierarchy of headings.

## Steps to reproduce

1. Add the Single Product block to a new post or page, select a grouped product and update the Add to Cart with Options block to the blockified version (_Add to Cart with Options (Experimental)_).
2. Notice the grouped product title is an h2:

![Image](https://github.com/user-attachments/assets/ff27298c-27ce-4c9b-accd-48133ce7d2b0)

3. Notice the children product titles are also an h2:

![Image](https://github.com/user-attachments/assets/e2396cac-12a8-41b0-8e8b-52260854e99d)

## Expected result

The children product titles should be an h3 for accessibility reasons. However, that should only be the case when the block is inside the Single Product block, not when used in the Single Product template, in which case the hierarchy of headings is fine.

## Relevant code

https://github.com/woocommerce/woocommerce/blob/038bfbee7e16b98903c0634af70af60fdfb74c05/plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-template/constants.ts#L35

**Comment by Aljullu** [comment]
Turns out this is a bit more tricky than I originally expected. We use the same template part for the _Single Product_ block and the _Single Product_ template, so we can't easily have different heading levels depending on the context.

After a quick discussion with @mdperez86, we realized that those product titles shouldn't be headings to start with. They would better be a `<p>` or a `<div>`/`<span>`. Given that the _Product Title_ block uses the _Post title_ block from Gutenberg, I think this issue is blocked by https://github.com/WordPress/gutenberg/issues/30549.

**Comment by Aljullu** [comment]
After discussing this issue with @sunyatasattva, we think a `<label>` might be a good candidate element too. There are some challenges (besides the block currently only allowing heading elements):

1. We would need to make sure the label is attached to the input next to it.
2. We would need to handle the case where an input is not rendered (for example, variable products or out of stock products).
3. Even though inner blocks are locked by default. Users can unlock them and remove them. So we would need to handle the case where the user removes the CTA block or the Product Title block.
